### PR TITLE
Object Storage change: Add CallBack API for mutli-part uploads - Issu…

### PR DESCRIPTION
**Pull request by srinivasthanneeru**
…e#187

### CODE REVIEW

File: objectstorage/transfer/file_uploader.go [hunk 1]
```diff
@@ -15,13 +15,13 @@ import (
 // FileUploader is an interface to upload a file
 type FileUploader interface {
 	// split file into multiple parts and uploads them to blob storage, then merge
-	UploadFileMultiparts(ctx context.Context, request UploadFileRequest) (response UploadResponse, err error)
+	UploadFileMultiparts(ctx context.Context, request UploadFileRequest, callBack objectstorage.UploadCallBack) (response UploadResponse, err error)
```
Comment (jasonyin) [inline objectstorage/transfer/file_uploader.go]: these public interface changes are breaking existing customers. Can you put callback inside request object and make it optional?

File: objectstorage/transfer/multipart_manifest.go [hunk 2]
```diff
@@ -17,14 +17,15 @@ type multipartManifest struct {
...
 type uploadPart struct {
-	size     int64
-	offset   int64
-	partBody []byte
-	partNum  int
-	hash     *string
-	opcMD5   *string
-	etag     *string
-	err      error
+	size       int64
```
Comment (jasonyin) [inline objectstorage/transfer/multipart_manifest.go]: looks like lots of formatting diffs. could you please run make test locally?

File: objectstorage/upload_part_request_response.go [hunk 3]
```diff
@@ -95,6 +96,23 @@ type UploadPartResponse struct {
 	ETag *string `presentIn:"header" name:"etag"`
 }
 
+//
+// MultiPartUploadPart holds the details of Part that is uploaded
+//
+type MultiPartUploadPart struct {
```
Comment (jasonyin) [inline objectstorage/upload_part_request_response.go]: any modification here will be overwritten by the generated code, please move it under transfer folder.

File: objectstorage/transfer/file_uploader.go (hunk 1 above)
Comment (srinivasthanneeru) [reply objectstorage/transfer/file_uploader.go]: ok, done

File: objectstorage/upload_part_request_response.go (hunk 3 above)
Comment (srinivasthanneeru) [reply objectstorage/upload_part_request_response.go]: sure, done

File: objectstorage/transfer/multipart_manifest.go (hunk 2 above)
Comment (srinivasthanneeru) [reply objectstorage/transfer/multipart_manifest.go]: sure, done

File: objectstorage/transfer/file_uploader.go (hunk 1 above)
Comment (srinivasthanneeru) [reply objectstorage/transfer/file_uploader.go]: sure, done

**Comment by jasonyin** [comment]
included in https://github.com/oracle/oci-go-sdk/releases/tag/v12.0.0

**Pull request closed without merging**